### PR TITLE
kubelet: move common types to kubelet/types

### PR DESCRIPTION
**Pull request by yujuhong**
This would faciliate tasks such as moving code in pkg/kubelet to sub packages.

### CODE REVIEW

**Comment by yujuhong** [comment]
@kubernetes/goog-node and @timstclair. Suggestions/objections are welcome.

File: cmd/integration/integration.go [hunk 1]
```diff
@@ -47,10 +47,10 @@ import (
 	"k8s.io/kubernetes/pkg/controller/node"
 	replicationControllerPkg "k8s.io/kubernetes/pkg/controller/replication"
 	"k8s.io/kubernetes/pkg/fields"
-	"k8s.io/kubernetes/pkg/kubelet"
 	"k8s.io/kubernetes/pkg/kubelet/cadvisor"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 	"k8s.io/kubernetes/pkg/kubelet/dockertools"
+	kubeletTypes "k8s.io/kubernetes/pkg/kubelet/types"
```
Comment (timstclair) [inline cmd/integration/integration.go]: I know this is consistent with what we already have, but what do you think about `kubetypes` instead? It's shorter and consistent with `kubecontainer`. Probably makes sense to change separately.

Comment (yujuhong) [inline cmd/integration/integration.go]: ok with kubetypes. Might as well do it in this PR. I'll wait to see if there is any other suggestions and address them together.

**Comment by timstclair** [comment]
LGTM. I just changed back to `kubeletTypes` in my ContainerID PR for better consistency >.<

**Comment by yujuhong** [comment]
Haha I will just rebase after yours and change them all to kubetype.

**Comment by yujuhong** [comment]
@dchen1107, @vishh, if there is no objection, I will update the PR based on @timstclair's suggestion.

**Comment by dchen1107** [comment]
kubeType is find with me.

**Comment by smarterclayton** [comment]
Package names and short names should never contain uppercase characters -
it's part of our conventions.

**Comment by smarterclayton** [comment]
For reference: https://blog.golang.org/package-names

**Comment by smarterclayton** [comment]
Specifically "When renaming an imported package, the local name should
follow the same guidelines as package names (lower case, no under_scores or
mixedCaps)."

**Comment by yujuhong** [comment]
> Specifically "When renaming an imported package, the local name should
> follow the same guidelines as package names (lower case, no under_scores or
> mixedCaps)."

Thanks! I will upload another PR to rename kubeletTypes to kubetypes. This would reduce the need to rebase this one :)

@timstclair, are you ok with a separate PR to address the naming?

**Comment by timstclair** [comment]
That's fine. Maybe fix everything while you're at it? :) 
https://gist.github.com/timstclair/d63aac28d4201cf0c0f0

**Comment by yujuhong** [comment]
> That's fine. Maybe fix everything while you're at it? :) 
> https://gist.github.com/timstclair/d63aac28d4201cf0c0f0

Fixing them is easy, but merging them is hard :(
I'll do what I can, but they probably need several PRs to fix.

**Comment by yujuhong** [comment]
@timstclair What about an LGTM? :)

**Comment by yujuhong** [comment]
Rebased. Re-applying lgtm.

**Pull request merged**